central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/stackrox/rox/central/graphql/resolvers"
+	"github.com/stackrox/rox/central/graphql/resolvers/loaders"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/branding"
+	"github.com/stackrox/rox/pkg/csv"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/notifier"
+	"github.com/stackrox/rox/pkg/notifiers"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/templates"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	reportGenCtx = resolvers.SetAuthorizerOverride(loaders.WithLoaderContext(sac.WithAllAccess(context.Background())), allow.Anonymous())
+
+	csvHeader = []string{
+		"Control Reference",
+		"Check(CCR)",
+		"Profile(version)",
+		"Check Description",
+		"Cluster",
+		"Status",
+		"Remediation",
+	}
+)
+
+const (
+	maxNumberProfilesinSubject = 4
+)
+
+type formatBody struct {
+	BrandedPrefix    string
+	Profile          string
+	Pass             int
+	Fail             int
+	Mixed            int
+	Cluster          int
+	ComplianceStatus string
+}
+
+type formatSubject struct {
+	BrandedPrefix string
+	ScanConfig    string
+	Profiles      string
+}
+
+type complianceReportGeneratorImpl struct {
+	checkResultsDS        checkResults.DataStore
+	notificationProcessor notifier.Processor
+}
+
+// struct which hold all columns of a row
+type resultRow struct {
+	ClusterName string
+	CheckName   string
+	Profile     string
+	ControlRef  string
+	Description string
+	Status      string
+	Remediation string
+}
+
+type resultEmail struct {
+	resultCSVs map[string][]*resultRow //map of cluster id to slice of *resultRow
+	totalPass  int
+	totalFail  int
+	totalMixed int
+	profiles   []string
+	clusters   int
+}
+
+func (rg *complianceReportGeneratorImpl) ProcessReportRequest(ctx context.Context, req *ComplianceReportRequest) error {
+
+	data, err := rg.getDataforReport(req, ctx)
+	if err != nil {
+		return err
+	}
+
+	zipData, err := rg.Format(data.resultCSVs)
+	if err != nil {
+		return err
+	}
+	profiles := strings.Join(req.profiles, ", ")
+	formatEmailBody := &formatBody{
+		BrandedPrefix: branding.GetCombinedProductAndShortName(),
+		Profile:       profiles,
+		Pass:          data.totalPass,
+		Fail:          data.totalFail,
+		Mixed:         data.totalMixed,
+		Cluster:       len(req.clusterIDs),
+	}
+	formatEmailBody.ComplianceStatus = "non compliant"
+	if data.totalFail == 0 {
+		formatEmailBody.ComplianceStatus = "compliant"
+	}
+
+	if len(profiles) > maxNumberProfilesinSubject {
+		profiles = strings.Join(req.profiles[0:maxNumberProfilesinSubject], ", ")
+		profiles += "..."
+	}
+
+	formatEmailSub := &formatSubject{
+		BrandedPrefix: branding.GetCombinedProductAndShortName(),
+		ScanConfig:    req.scanConfigName,
+		Profiles:      profiles,
+	}
+
+	return rg.sendEmail(zipData, formatEmailBody, formatEmailSub, req.notifiers, ctx)
```

Comment:
We need to send the email asynchronously and not wait for the send to complete - I think we need to use a go routine in sendEmail to just post it to the notifier and return